HIP/hip-513.md

```diff
@@ -421,6 +421,31 @@ nodes that enable sidecar functionality. Below is an example of this state chang
 }
 ```
 
+#### Contract State REST API
+
+While the state changes above shows how particular slot values change as a result of a contract invocation, it does not provide a complete list of all slots and their values. For that, 
+we add an `/api/v1/contracts/{idOrAddress}/state` REST API where `id` is either the `shard.realm.num`, `realm.num`, `num`, or a hex encoded EVM address of the contract.
+
+```json
+{
+  "state": [{
+    "address": "0x0000000000000000000000000000000000001f41",
+    "contract_id": "0.0.100",
+    "slot": "0x",
+    "timestamp": "10000000.000000001",
```

Comment:
Will we be looking to support time based queries i.e. `/api/v1/contracts/{idOrAddress}/state?slot=0x&timestamp=T1`
This would allow one to query the value of a slot at a given time